docs/pkg/client/events.md

```diff
@@ -0,0 +1,391 @@
+  - [Why use `events` over directly using Gorilla WebSockets?](#why-use-events-over-directly-using-gorilla-websockets)
+  - [How can I use a different connection mechanism other than WebSockets?](#how-can-i-use-a-different-connection-mechanism-other-than-websockets)
+  - [Why use the `EventsReplayClient` over directly maintaining an `EventsQueryClient`?](#why-use-the-eventsreplayclient-over-directly-maintaining-an-eventsqueryclient)
+
+<!-- tocstop -->
+
+## Overview
+
+The `events` package provides a client interface to subscribe to chain event
+messages. It abstracts the underlying connection mechanisms and offers a clear
+and easy-to-use way to get events from the chain. Highlights:
+
+- Offers subscription to chain event messages matching a given query.
+- Uses the Gorilla WebSockets package for underlying connection operations.
+- Provides a modular structure with interfaces allowing for mock implementations
+  and testing.
+- Offers considerations for potential improvements and replacements, such as
+  integration with the cometbft RPC client.
+- Offers a generic client to decode on chain event bytes into the desired event
+  type
+
+## Architecture Diagrams
+
+### Components
+
+```mermaid
+---
+title: Component Diagram Legend
+---
+
+flowchart
+
+  a[Component A]
+  b[Component B]
+  c[Component C]
+  d[Component D]
+
+  a --"A uses B via B#MethodName()"--> b
+  a =="A returns C from A#MethodName()"==> c
+  b -."B uses D via network IO".-> d
+```
+
+```mermaid
+---
+title: EventsQueryClient Components
+---
+
+flowchart
+
+  subgraph comet[Cometbft Node]
+    subgraph rpc[JSON-RPC]
+      sub[subscribe endpoint]
+    end
+  end
+
+  subgraph eqc[EventsQueryClient]
+    q1_eb[EventsBytesObservable]
+    q1_conn[Connection]
+    q1_dial[Dialer]
+  end
+
+  q1_obsvr1[Observer 1]
+  q1_obsvr2[Observer 2]
+
+
+  q1_obsvr1 --"#Subscribe()"--> q1_eb
+  q1_obsvr2 --"#Subscribe()"--> q1_eb
+
+
+  q1_dial =="#DialContext()"==> q1_conn
+  q1_eb --"#Receive()"--> q1_conn
+
+  q1_conn -.-> sub
+```
+
+```mermaid
+---
+title: EventsReplayClient Components
+---
+
+flowchart
+  subgraph chain[Pocket Network]
+    c1_md[Action]
+  end
+
+  subgraph erc[EventsReplayClient]
+    subgraph eqc[EventsQueryClient]
+      r1_qs[Event Query String]
+      r1_qe[Websocket URL]
+    end
+    r1_rt(retryPublishEventsFactory)
+    subgraph dec[EventBytesDecoder]
+      r1_df[Decoder Function]
+    end
+    r1_ob[LatestReplayObservable]
+    r1_pb[LatestReplayObservablePublisher]
+  end
+
+  subgraph con[Consumer]
+    c1_ob[Replay Observable]
+    c1_ev[Events]
+  end
+
+  c1_md -."#EmitTypedEvent".-> eqc
+
+  eqc --"retry.OnError"--> r1_rt
+  r1_df --"Decoded Event"--> r1_rt
+  r1_rt -."#EventBytes".-> eqc
+  r1_rt --"LatestReplayObservable"-->r1_pb
+  r1_pb --> r1_ob
+
+  eqc --"Serialised Event"--> r1_df
+  r1_ob --"#EventsSequence"--> c1_ob
+  r1_ob --"#LastNEvents"--> c1_ev
+```
+
+### Subscriptions
+
+```mermaid
+---
+title: Event Subscription Data Flow
+---
```

Comment:
This diagram needs one of:

A. Its own legend, in which the usage via network I/O association direction matches the diagram
B. To be rewritten in terms of the legend provided in the previous section (i.e. turn it upside-down)
C. A note or some other indicator that the legend in the section above only applies to that section and/or does not apply to this section.

I think A is ideal but C would be sufficient for now. I think this diagram reads better as written (as opposed to upside-down). I also think that the component diagrams above read best  they way they are (i.e. shouldn't be turned upside-down either).

Reply:
Added todos to add legends in the appropriate places